### PR TITLE
Update SocketReader::ReadSome SocketError when socket closed

### PR DESCRIPTION
**Pull request by oclancy**
Update SocketInitiatorThread::ReadSome SocketError when socket closed

If 0 bytes are read it means the socket has been shutdown not reset. Had this issue in a production system reporting "Connection reset by peer" (the error message for SocketError.SocketReset), but using wireshark no RST packet was seen. However we did see the FIN and FIN,ACK packets which means the socket was being deliberately shutdown. Turned out the venue had a maintenance job that shutdown the network connections.

Tldr; Setting to SocketError.Shutdown is the correct message for this scenario.

### CODE REVIEW

**Comment by oclancy** [comment]
Apologies - thought I'd already signed it.

**Comment by gbirchmeier** [comment]
why closed?

**Comment by oclancy** [comment]
Sorry I had to reopen it [here ](https://github.com/connamara/quickfixn/pull/785)- I screwed up my fork a bit.

**Comment by gbirchmeier** [comment]
👍
